packages/main/src/Calendar.ts

```diff
@@ -789,18 +788,66 @@ class Calendar extends CalendarPart {
 	}
 
 	get accInfo() {
+		const currentYearRange = this._currentYearRange;
+		const { rangeStartText, rangeEndText } = this._formatYearRangeText(currentYearRange);
+
+		const headerMonthButtonText = this.hasSecondaryCalendarType
+			? `${this._headerMonthButtonText}, ${this.secondMonthButtonText}` : `${this._headerMonthButtonText}`;
+
+		// Get base labels
+		const monthLabel = Calendar.i18nBundle?.getText(CALENDAR_HEADER_MONTH_BUTTON, headerMonthButtonText);
+		const yearLabel = Calendar.i18nBundle?.getText(CALENDAR_HEADER_YEAR_BUTTON, this._headerYearButtonText as string);
+		const yearRangeLabel = Calendar.i18nBundle?.getText(CALENDAR_HEADER_YEAR_RANGE_BUTTON, rangeStartText, rangeEndText);
+
+		// Get shortcuts
+		const monthShortcut = Calendar.i18nBundle?.getText(CALENDAR_HEADER_MONTH_BUTTON_SHORTCUT);
+		const yearShortcut = Calendar.i18nBundle?.getText(CALENDAR_HEADER_YEAR_BUTTON_SHORTCUT);
+		const yearRangeShortcut = Calendar.i18nBundle?.getText(CALENDAR_HEADER_YEAR_RANGE_BUTTON_SHORTCUT);
+
 		return {
-			ariaLabelMonthButton: this.hasSecondaryCalendarType
-				? `${this._headerMonthButtonText}, ${this.secondMonthButtonText}` : `${this._headerMonthButtonText}`,
+			ariaLabelMonthButton: monthLabel,
+			ariaLabelYearButton: yearLabel,
+			ariaLabelYearRangeButton: yearRangeLabel,
+
+			// Keyboard shortcuts for aria-keyshortcuts
+			keyShortcutMonthButton: monthShortcut,
+			keyShortcutYearButton: yearShortcut,
+			keyShortcutYearRangeButton: yearRangeShortcut,
+
+			// Tooltips combining label and shortcut
+			tooltipMonthButton: `${monthLabel} (${monthShortcut})`,
+			tooltipYearButton: `${yearLabel} (${yearShortcut})`,
+			tooltipYearRangeButton: `${yearRangeLabel} (${yearRangeShortcut})`,
 		};
 	}
 
-	get headerPreviousButtonText() {
-		return Calendar.i18nBundle?.getText(CALENDAR_HEADER_PREVIOUS_BUTTON);
+	/**
+	 * Helper method to create CalendarDateComponent instances for year range
+	 * @private
+	 */
+	_createYearRangeDates(yearRange: CalendarYearRangeT, calendarType: string = this._primaryCalendarType) {
+		const rangeStart = new CalendarDateComponent(this._calendarDate, calendarType);
+		const rangeEnd = new CalendarDateComponent(this._calendarDate, calendarType);
+
+		rangeStart.setYear(yearRange.startYear);
+		rangeEnd.setYear(yearRange.endYear);
+
+		return { rangeStart, rangeEnd };
 	}
 
-	get headerNextButtonText() {
-		return Calendar.i18nBundle?.getText(CALENDAR_HEADER_NEXT_BUTTON);
+	/**
+	 * Helper method to format year range text
+	 * @private
+	 */
+	_formatYearRangeText(yearRange: CalendarYearRangeT, calendarType?: string) {
+		const actualCalendarType = calendarType || this.primaryCalendarType;
```

Comment:
we usually try to use the word "effective" for such cases, so effectiveCalendarType